Update repository documentation and training exercises

AlexGisi opened the pull request.

AlexGisi committed:
Improve documentation README

AlexGisi committed:
Update requirements.txt

AlexGisi committed:
Add VS Code info to setup docs

AlexGisi committed:
Fix/improve 3.1-3.3

AlexGisi committed:
Add venv/

AlexGisi committed:
Fix/minor improvement

DavidMerzJr commented on gh_pages/README.md:
````diff
@@ -2,18 +2,29 @@
+    cd <industrial_training root directory>
+    python3 -m venv/ venv/
+    source venv/bin/activate
+    ```
````
Small suggestion: find some way to mark that these three commands are the 'recommended'/optional ones, but the rest are required either way.  A box around it, or a hide tag, or something like that.  Otherwise, it may be unclear to the average reader which ones are required and which are not.

DavidMerzJr commented on gh_pages/_source/session3/ros2/2-Coordinate-Transforms-using-TF.md:
```diff
@@ -49,7 +49,7 @@ Specifically, edit the service callback inside the vision_node to transform the
     {
     public:
-      Localizer() : Node("vision_node"), buffer_(this->get_clock()), listener_(buffer_)
+      Localizer() : Node("vision_node"), last_msg_{nullptr}, buffer_(this->get_clock()), listener_(buffer_)
```
By default, C++ shared pointers will initialize with a `nullptr`, no action needed.  Is there a particular reason this was added here (build warning, etc.)?

DavidMerzJr commented on gh_pages/_source/session3/ros2/3-Build-a-MoveIt-Package.md:
````diff
@@ -101,7 +106,7 @@ In this exercise, you will generate a MoveIt package for the UR5 workcell you bu
     ```
 
- 1. Open the ROS2 `config/ompl_planning.yaml` file in a text editor and add the following lines at the bottom:
+ 1. Open the ROS2 `config/ompl_planning.yaml` file in a text editor and add the following lines at the bottom, with no indentation:
````
The "with no indentation" piece could be confusing - we do need some of these lines to have indentation.  Do you feel like "Make sure your editor does not add extra whitespace to the front of these lines" would have added the clarity you needed?

DavidMerzJr commented on gh_pages/_source/session5/Building-a-Perception-Pipeline.rst:
```diff
@@ -153,8 +153,15 @@ Viewing Results
+
+   .. code-block:: bash
+      cd ~/perception_ws
+      ros2 run lesson_perception pcd_to_pointcloud --ros-args -p filename:=table.pcd -p tf_frame:=kinect_link -p topic:=/kinect/depth_registered/points
```
Did `~/table.pcd` not work?

AlexGisi replied:
No -- somehow it gives `[pcl::PCDReader::read] Could not find file '~/table.pcd'.`

AlexGisi replied:
No -- somehow it gives `[pcl::PCDReader::read] Could not find file '~/table.pcd'.`

gavanderhoorn replied:
`${HOME}/table.pcd` might work instead.

Bash needs to perform the substitution, but it looks like it isn't doing it before passing on all CLAs to `ros2 run`.

DavidMerzJr commented on gh_pages/_source/setup/PC-Setup---ROS.md:
```diff
@@ -41,3 +41,49 @@ After this step (or if you already have a working ROS environment), clone the tr
 
 `~/industrial_training/.check_training_config.bash`
+
+## VS Code Integration
```
Instead of putting this IDE in PC Setup, I would prefer to see a separate page that walks through a few different IDEs (VS Code, Qt Creator) and how to install them.

In particular, saying to install VS Code here would conflict with the actual training workshops we put on, in which we (after this point) have people install Qt Creator.

DavidMerzJr reviewed (commented):
Thanks for your suggestions & fixes.  I had a few comments.

AlexGisi committed:
Make new page for IDE setup with QT Creator section

AlexGisi committed:
Demarcate optional portion of build instructions

AlexGisi committed:
Remove unnecessary explicit initialization of shared pointer

AlexGisi committed:
Clarify whitespace comment

AlexGisi commented:
Thanks David. I addressed these  in turn with the above commits.

AlexGisi committed:
Put table.pcd in home directory

DavidMerzJr commented on gh_pages/README.md:
````diff
@@ -1,19 +1,34 @@
+    ```
+
+</details>
+<hr>
````
This should make this look less like a section header:
```suggestion
    <hr>
    <details>
    <summary>(Recommended) Create a python virtual environment for Sphinx packages and dependencies:</summary>
 
        ```
        cd <industrial_training root directory>
        python3 -m venv/ venv/
        source venv/bin/activate
        ```

    </details>
    <hr>
```

DavidMerzJr commented on gh_pages/_source/session3/ros2/3-Build-a-MoveIt-Package.md:
````diff
@@ -113,6 +118,8 @@ In this exercise, you will generate a MoveIt package for the UR5 workcell you bu
     start_state_max_bounds_error: 0.1
     ```
+    
+    Make sure your editor does not add extra whitespace to the front of these lines.
````
```suggestion
    Make sure your editor does not change the amount of whitespace at the front of these lines.
```

DavidMerzJr commented on gh_pages/_source/setup/IDE-Setup.md:
```diff
@@ -0,0 +1,53 @@
+> An integrated development environment (IDE) is a text editor with a set of tools for helping you program, like syntax highlighting, autocomplete, and debugging. This section details the installation and setup of two IDEs commonly used for ROS development.
+
+## QT Creator
+[QT Creator](https://github.com/qt-creator/qt-creator) is a popular cross-platform IDE. ROS-Industrial provides a [ROS QT Creator plug-in](https://github.com/ros-industrial/ros_qtc_plugin) with features like Catkin integration, automatic sourcing of workspaces, and templates for commonly created files.
```
```suggestion
[QT Creator](https://github.com/qt-creator/qt-creator) is a popular cross-platform IDE. ROS-Industrial provides a [ROS QT Creator plug-in](https://github.com/ros-industrial/ros_qtc_plugin) with features like ROS build tool integration, automatic sourcing of workspaces, and templates for commonly created files.
```

DavidMerzJr commented on gh_pages/_source/setup/IDE-Setup.md:
```diff
@@ -0,0 +1,53 @@
+## QT Creator
+[QT Creator](https://github.com/qt-creator/qt-creator) is a popular cross-platform IDE. ROS-Industrial provides a [ROS QT Creator plug-in](https://github.com/ros-industrial/ros_qtc_plugin) with features like Catkin integration, automatic sourcing of workspaces, and templates for commonly created files.
+
+You can easily install the editor bundled with the plug-in from the Snap store [here](https://snapcraft.io/qtcreator-ros). Click the green `Install` button in the top right and click `View in Desktop store`. In the window which opens, install the application. Then you can open the IDE by searching for `QT Creator ROS` on your system.
```
Users of older systems will need to use the installer files from here: https://ros-qtc-plugin.readthedocs.io/en/latest/_source/How-to-Install-Users.html

AlexGisi replied:
Does this refer to <16.04? According to the snap page people are using the snap distribution down to 16.04. If so, does this just require a brief mention, since the vast majority of users can download from the snap store?

DavidMerzJr commented on gh_pages/_source/setup/IDE-Setup.md:
```diff
@@ -0,0 +1,53 @@
+[QT Creator](https://github.com/qt-creator/qt-creator) is a popular cross-platform IDE. ROS-Industrial provides a [ROS QT Creator plug-in](https://github.com/ros-industrial/ros_qtc_plugin) with features like Catkin integration, automatic sourcing of workspaces, and templates for commonly created files.
+
+You can easily install the editor bundled with the plug-in from the Snap store [here](https://snapcraft.io/qtcreator-ros). Click the green `Install` button in the top right and click `View in Desktop store`. In the window which opens, install the application. Then you can open the IDE by searching for `QT Creator ROS` on your system.
+
```
If you're adding the instructions for setting up a workspace in VS Code, please add the instructions for setting up a workspace in QTCreator

DavidMerzJr commented on gh_pages/_source/setup/IDE-Setup.md:
````diff
@@ -0,0 +1,53 @@
+```
+
+where `foxy` can be changed to the appropriate distro.
+
+Now you're prepared to work on ROS projects in VS Code.
````
```suggestion

```

DavidMerzJr replied:
We may want to add a disclaimer to each - let them know that the training assumes they will be using QT Creator, but if they are already familiar with VS Code then they *can* use it.  (They'll just be on their own for troubleshooting it.)

DavidMerzJr commented on gh_pages/_source/setup/IDE-Setup.md:
```diff
@@ -0,0 +1,53 @@
+You can easily install the editor bundled with the plug-in from the Snap store [here](https://snapcraft.io/qtcreator-ros). Click the green `Install` button in the top right and click `View in Desktop store`. In the window which opens, install the application. Then you can open the IDE by searching for `QT Creator ROS` on your system.
+
+## VS Code
+[Visual Studio Code](https://code.visualstudio.com/download) is another useful editor for ROS-related files. Download the editor and install the C/C++ extension pack and the ROS extension pack. Both are authored by Microsoft. 
```
```suggestion
[Visual Studio Code](https://code.visualstudio.com/download) is a Microsoft editor which can be used for ROS projects. Download the editor and install the C/C++ extension pack and the ROS extension pack. (Make sure you get the right ones - both extensions are authored by Microsoft.)
```

DavidMerzJr reviewed (commented):
A few more changes, for the most part I've made the changes for you to include.

There's a bit of nuance here - since most of our training staff is familiar with QT Creator and not VS Code, we want to make sure we are at least (if not more) helpful with setting up QT Creator.

AlexGisi committed:
Fix universal_robot commit

Previously used commit had unnecessary headers which made the build fail on certain machines

AlexGisi committed:
Standard markdown doesn't support lettered lists

AlexGisi committed:
Update gh_pages/README.md indentation

Co-authored-by: DavidMerzJr <dmmjr2@gmail.com>

AlexGisi committed:
Clarify whitespace suggestion

Co-authored-by: DavidMerzJr <dmmjr2@gmail.com>

AlexGisi committed:
Clarify QT build tool integration

Co-authored-by: DavidMerzJr <dmmjr2@gmail.com>

AlexGisi committed:
Update gh_pages/_source/setup/IDE-Setup.md

Co-authored-by: DavidMerzJr <dmmjr2@gmail.com>

AlexGisi committed:
Update gh_pages/_source/setup/IDE-Setup.md

Co-authored-by: DavidMerzJr <dmmjr2@gmail.com>

AlexGisi committed:
Recommend QT Creator

AlexGisi committed:
Add QT Creator project setup instructions
